### ISSUE TITLE
Add the ability to set the MAC address when starting a container

### ISSUE THREAD

**Issue by dcheese** [jenkinsci/docker-plugin#226]
We have a docker image that requires a static MAC address due to a node lock license requirement. I have tried to use the LXC Conf Options to set the MAC address with no luck. 

I changed the daemon to use the lxc driver (--exec-driver=lxc) and set the LXC option to
lxc.network.hwaddr=02:d0:c6:0a:29:33

It seems to just be ignored, so we went back to using the libcontainer driver (default)

So, to try to fix this,  I upgraded my fork to use version 1.3 of the docker-java dependency which introduced a withMacAddress method to the dockerjava/api/command/CreateContainerCmd.java class.

When I modify the DockerTemplateBase class of the plugin and set containerConfig.withMacAddress("02:d0:c6:0a:29:33"), the container starts with that MAC Address. 

So, here are my questions:

    1) Is there a way to set the container before starting to a static MAC address with the current plugin
         version? 
    2) If the answer to 1 is no, are there any plans to upgrade the plugin to use version 1.3 of docker-java?
    3) If the answer to 2 is yes, would adding a MAC Address field to the plugin config panel make sense?

Basically, we really need a way to set a static MAC address when starting a container from the plugin configuration panel that works.

**Comment by KostyaSha** [comment]
> static MAC address due to a node lock license requirement

How amazing this world...

**Comment by dcheese** [comment]
I know, in our case, it because InstallAnywhere license is tied to a MAC address and we want to be able to run that from a docker image.

So, is this something that I should be able to do with the current plugin version?

**Comment by KostyaSha** [comment]
Please verify this hpi https://jenkins.ci.cloudbees.com/job/plugins/job/docker-plugin/286/com.nirima$docker-plugin/ and reply
